#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/CircleArena.proto"

WorldInfo {
}
Viewpoint {
  orientation 0.2331045680639751 -0.28168846442511825 -0.9307598344135434 4.829667572342943
  position -0.8375415525479062 -1.4213618851704157 1.0348103493047822
}
TexturedBackground {
}
TexturedBackgroundLight {
}
TexturedBackgroundLight {
}
CircleArena {
}
Robot {
  translation -0.6 0 0.03
  children [
    DEF RL1 HingeJoint {
      jointParameters HingeJointParameters {
        position 1.8196104787637608e-11
        axis 0 0 1
        anchor -0.0325 0.00775 0.27
      }
      device [
        RotationalMotor {
          name "rl1_motor"
        }
        PositionSensor {
          name "rl1_pos"
        }
      ]
      endPoint Solid {
        translation -0.0325 0.007749999992630577 0.27
        children [
          DEF RL2 HingeJoint {
            jointParameters HingeJointParameters {
              axis 0 1 0
              anchor 0 0.025 -0.03
            }
            device [
              RotationalMotor {
                name "rl2_motor"
              }
              PositionSensor {
                name "rl2_pos"
              }
            ]
            endPoint Solid {
              translation 0 -4.896777427987331e-14 -0.03
              rotation 1 0 0 1.5708
              children [
                DEF RL3 HingeJoint {
                  jointParameters HingeJointParameters {
                  }
                  device [
                    RotationalMotor {
                      name "rl3_motor"
                    }
                    PositionSensor {
                      name "rl3_pos"
                    }
                  ]
                  endPoint Solid {
                    rotation 0 1 0 1.5708053071795867
                    children [
                      DEF Upper_Leg_Servo Solid {
                        translation 0 -0.1125 0
                        rotation 0 1 0 0
                        children [
                          DEF RL4 HingeJoint {
                            jointParameters HingeJointParameters {
                              position 0.030720000000000004
                              axis 0 0 1
                              anchor 0.000307 -0.01 0
                            }
                            device [
                              RotationalMotor {
                                name "rl4_motor"
                              }
                              PositionSensor {
                                name "rl4_pos"
                              }
                            ]
                            endPoint Solid {
                              translation 1.4484938244777599e-07 -0.010009429556695657 0
                              rotation 0 0 1 0.030720000000002978
                              children [
                                DEF Lower_Leg_Servo Solid {
                                  translation 0 -0.09 0
                                  children [
                                    DEF rl5_servo_box Shape {
                                      appearance PBRAppearance {
                                        baseColor 0 0 0
                                        roughness 1
                                        metalness 0
                                      }
                                      geometry Box {
                                        size 0.03 0.03 0.03
                                      }
                                    }
                                    DEF RL5 HingeJoint {
                                      jointParameters HingeJointParameters {
                                        axis 0 0 1
                                        anchor -0.002 -0.005 0
                                      }
                                      device [
                                        RotationalMotor {
                                          name "rl5_motor"
                                        }
                                        PositionSensor {
                                          name "rl5_pos"
                                        }
                                      ]
                                      endPoint Solid {
                                        translation -0.0020000000000034712 -0.005000000000005694 -1.7069679003611782e-15
                                        children [
                                          DEF rl6_servo Shape {
                                            appearance PBRAppearance {
                                              baseColor 1 0 0
                                              roughness 1
                                              metalness 0
                                            }
                                            geometry Cylinder {
                                              height 0.05
                                              radius 0.001
                                            }
                                          }
                                          DEF rl6_servo_box Shape {
                                            appearance PBRAppearance {
                                              baseColor 0 0 0
                                              roughness 1
                                              metalness 0
                                            }
                                            geometry Box {
                                              size 0.02 0.03 0.03
                                            }
                                          }
                                          DEF RL6 HingeJoint {
                                            jointParameters HingeJointParameters {
                                            }
                                            device [
                                              RotationalMotor {
                                                name "rl6_motor"
                                              }
                                              PositionSensor {
                                                name "rl6_pos"
                                              }
                                            ]
                                            endPoint Solid {
                                              rotation -2.2243223143017736e-11 1 -2.224330484708891e-11 1.5708
                                              children [
                                                DEF Right_Foot Transform {
                                                  translation 0.0350001 -0.0300001 0.00525001
                                                  rotation 1.0171504044775673e-08 -0.7071062811861719 -0.7071072811865696 -3.1415853071795863
                                                  scale 0.001 0.001 0.001
                                                  children [
                                                    DEF right_foot Shape {
                                                      appearance PBRAppearance {
                                                        baseColor 0 0 1
                                                        roughness 1
                                                        metalness 0
                                                      }
                                                      geometry Mesh {
                                                        url [
                                                          "STL_files/left_foot.stl"
                                                        ]
                                                      }
                                                    }
                                                  ]
                                                }
                                              ]
                                              physics Physics {
                                              }
                                            }
                                          }
                                        ]
                                        boundingObject USE rl6_servo_box
                                        physics Physics {
                                        }
                                      }
                                    }
                                  ]
                                  boundingObject USE rl5_servo_box
                                  physics Physics {
                                  }
                                }
                                DEF Lower_Leg Transform {
                                  translation 5.6553e-08 -0.125 -0.0385
                                  rotation 0.5773492691885823 0.5773502691890485 0.5773512691895147 -2.094395307179586
                                  scale 0.001 0.001 0.001
                                  children [
                                    DEF lower_leg Shape {
                                      appearance PBRAppearance {
                                        baseColor 0 0 1
                                        roughness 1
                                        metalness 0
                                      }
                                      geometry Mesh {
                                        url [
                                          "STL_files/lower_leg.stl"
                                        ]
                                      }
                                    }
                                  ]
                                }
                                DEF rl4_servo Shape {
                                  appearance PBRAppearance {
                                    baseColor 1 0 0
                                    roughness 1
                                    metalness 0
                                  }
                                  geometry Cylinder {
                                    height 0.05
                                    radius 0.001
                                  }
                                }
                              ]
                              boundingObject USE rl4_servo
                              physics Physics {
                              }
                            }
                          }
                          DEF rl4_servo_box Shape {
                            appearance PBRAppearance {
                              baseColor 0 0 0
                              roughness 1
                              metalness 0
                            }
                            geometry Box {
                              size 0.022 0.04 0.04
                            }
                          }
                        ]
                        boundingObject USE rl4_servo_box
                        physics Physics {
                        }
                      }
                      DEF Upper_Leg Transform {
                        translation -0.00249997 -0.2475 -0.0375
                        rotation 0.577349935856137 0.577349935856137 0.5773509358560258 -2.094395307179586
                        scale 0.001 0.001 0.001
                        children [
                          DEF r_upper_leg Shape {
                            appearance PBRAppearance {
                              baseColor 0 0 1
                              roughness 1
                              metalness 0
                            }
                            geometry Mesh {
                              url [
                                "STL_files/left_upper_leg.stl"
                              ]
                            }
                          }
                        ]
                      }
                      DEF rl3_servo Shape {
                        appearance PBRAppearance {
                          baseColor 1 0 0
                          roughness 1
                          metalness 0
                        }
                        geometry Cylinder {
                          height 0.04
                          radius 0.001
                        }
                      }
                    ]
                    boundingObject USE rl3_servo
                    physics Physics {
                    }
                  }
                }
                DEF rl3_servo_box Shape {
                  appearance PBRAppearance {
                    baseColor 0 0 0
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 0.03 0.033 0.03
                  }
                }
                DEF rl2_servo Shape {
                  appearance PBRAppearance {
                    baseColor 1 0 0
                    roughness 1
                    metalness 0
                  }
                  geometry Cylinder {
                    height 0.005
                    radius 0.005
                  }
                }
              ]
              boundingObject USE rl2_servo
              physics Physics {
              }
            }
          }
          DEF rl1_servo_bracket Transform {
            translation -0.345 -0.0075 0.0700008
            rotation 0 1 0 1.5708
            scale 0.001 0.001 0.001
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0 0 0
                  roughness 1
                  metalness 0
                }
                geometry Mesh {
                  url [
                    "STL_files/left_shoulder.stl"
                  ]
                }
              }
            ]
          }
          DEF rl1_servo Shape {
            appearance PBRAppearance {
              baseColor 0 0 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.01
              radius 0.005
            }
          }
        ]
        name "solid(1)"
        boundingObject USE rl1_servo
        physics Physics {
        }
      }
    }
    DEF Back_Plate Transform {
      scale 0.001 0.001 0.001
      children [
        DEF Chest_Back Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 1
            metalness 0
          }
          geometry Mesh {
            url [
              "STL_files/chest_back.stl"
            ]
          }
        }
      ]
    }
    DEF Chest_Plate Transform {
      scale 0.001 0.001 0.001
      children [
        DEF Chest Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 1
            metalness 0
          }
          geometry Mesh {
            url [
              "STL_files/Chest.stl"
            ]
          }
        }
      ]
    }
  ]
  name "robot(2)"
  physics Physics {
  }
  controller "leg_tutorial"
}
